contracts/solidity/NFTXVaultListingUpgradeable.sol

```diff
@@ -0,0 +1,307 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity ^0.8.0;
+
+import "./interface/INFTXVault.sol";
+import "./interface/INFTXVaultFactory.sol";
+import "./interface/INFTXVaultListing.sol";
+import "./interface/INFTXFeeDistributor.sol";
+import "./testing/IERC721.sol";
+import "./util/ReentrancyGuardUpgradeable.sol";
+
+import "hardhat/console.sol";
+
+// Authors: @tomwade.
+
+contract NFTXVaultListingUpgradeable is ReentrancyGuardUpgradeable, INFTXVaultListing {
+
+	struct Listing {
+		uint id;
+		uint vaultId;
+		address vaultAddress;
+		uint nftId;
+		uint price;
+		bool active;
+		uint expiry;
+		address seller;
+	}
+
+    event ListingCreated();
```

Comment:
`(uint256 listingId, uint256 nftId, address vault, uint256 price, uint256 expires)`

I don't believe we can do an event with all the data so would need to loop through each listing created and emit an event.